promiseWrapper.js

```diff
@@ -1,138 +1,107 @@
-//   });
-// };
-
-// connect(options).then((string) => {
-//   console.log('amazing');
-//   console.log(string);
-// });
+'use strict';
 
+const shellEscape = require('shell-escape');
+const debug = require('debug')('promiseWrapper');
 
 class SSH {
+  /**
+   * Constructor, called with options object that contains host, username and password
+   * for the connection. Does not connect to the host.
+   *
+   * new SSH(options: host, username, password)
+   * @returns {*|Promise.<TResult>}
+   */
   constructor(options) {
+    debug('created promise wrapper object');
+    this.connected = false;
     this.connection = null;
     this.options = options;
     this.connection = require('./lib/index.js')(this.options);
-    this.connection.on('stderr', function(err) { console.log("Error on StdErr",err); });
-    this.connection.once('closed', function(addr,err) {
-	    if (err) {
-		    console.log("Error on session",err);
-	    }else { console.log("SSH Session has being closed."); }
-    });
+    this.connection.on('stderr', function(err) { console.error('Error on StdErr',err); });
   }
+
+  /**
+   * Connect to the host specified in the constructor
+   *
+   * connect()
+   * @returns {*|Promise.<Tresult>}
+   */
   connect() {
     return new Promise((resolve, reject) => {
       this.connection.on('closed', reject);
       this.connection.on('connected',(host) => {
+        debug('Connecting to host:', host);
         this.connection.removeListener('closed', reject);
-        console.log('Were Connected',host);
-        resolve();
+        this.connected = true;
+        resolve('Connected to host:', host);
       });
       this.connection.connect();
     });
   }
-  write(value) {
+
+  /**
+   * Run a command 'value' on the remote SSH connection, requires active connection
+   * @param value {command to be run}
+   * @param options {Options object, contains cwd (directory to execute value from)}
+   * @returns {*|Promise< stdout: string>}
+   */
+  execCommand(value, options) {
     return new Promise((resolve, reject) => {
-      console.log('in promise')
+      if (options.cwd) {
+        value = `cd ${shellEscape([options.cwd])} 1> /dev/null 2> /dev/null; ${value}`;
+      }
+      let wholeLine = '';
       this.connection.on('output', (data,lastChunk) => {
-        console.log('in event')
 	      if (data) {
-		      if (data.match("pwd")) {
-            console.log("Command data->",data);
-		      }else {
-            console.log("Output data->",data);
-            resolve();
-          }
-	      }
+          wholeLine += data;
+        } else {
+          if (lastChunk === true) {
+            let res = '';
+            this.connection.removeAllListeners('output');
+            if (wholeLine.indexOf(value) !== -1) {
+              debug('$', value);
+              res = wholeLine.replace(value, '');
+            } else {
+              console.error(`Expected ${value} to be present in ${wholeLine}`);
```

Comment:
We could still debug to log this error message. However, with this approach an error will get logged regardless of the DEBUG content so I reckon this is a good approach.